### PR TITLE
Use get_last_lr() method on schedulers

### LINKED ISSUE #5363: Use scheduler.get_last_lr() instead of manually searching for optimizers.param_groups
## 🐛 Bug / Feature Request
I am not sure whether to classify this issue as a feature request or bug, but in your codebase, you get the learning rate from schedulers as ``scheduler.optimizer.param_groups[0]['lr']``.
However, PyTorch provides the ``get_last_lr()`` method for this.

I created my own scheduler, which allows me to combine multiple schedulers together (eg. burn-in LR for the first few steps, and then step down LR later during training). 
As this scheduler holds multiple different schedulers itself, there is no ``scheduler.optimizer`` argument and thus I cannot use this with PyTorch Lightning.
I did however implement the ``get_last_lr()`` method, which gets the LR from the underlying schedulers, so if you could modify the PyTorch Lightning code to use that method, it should all work!

## Problems
Obviously, nothing is ever as straight forward with the PyTorch scheduler API, which means that there might be problems with this solution. One small issue I found is that ``ReduceLROnPlateau`` is behaving differently and does not implement the ``get_last_lr()`` method. (The PyTorch team should really look into unifying their API, but that should be discussed there...) 

If you look at GH-5364, which I made on top of the `release/1.2-dev` branch, you will see that for the ``ReduceLROnPlateau`` scheduler, I still use the old method of finding the LR through the ``scheduler.optimizer`` object.
However, for any other scheduler, we use the ``get_last_lr()`` method.

I could adapt the code to be even more defensive and check whether there is a ``get_last_lr()`` method and only use it if it is available, but I don't know whether that is necessary, as all other official schedulers have the same base Scheduler class which implements this method.

---

I send in a Draft PR with the necessary changes.
Kind Regards,
0phoff

### PR DESCRIPTION
**Pull request by 0phoff**
## What does this PR do?
Instead of manually looking for the LR through ``scheduler.optimizer.param_groups[0]['lr']``, we can use the ``get_last_lr()`` method which is available on all Schedulers (except for ReduceLROnPlateau, which is handled separately).

Fixes #5363

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [X] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [X] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [X] Did you make sure to update the documentation with your changes [if needed]?
- [ ] Did you write any new necessary tests [no need for typos, docs]? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [X] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [ ] **Check that target branch and milestone are aligned!**
 

## Did you have fun?
Make sure you had fun coding 🙃
:heavy_check_mark:

### CODE REVIEW

**Comment by 0phoff** [comment]
@Borda, I only made some minor changes which I think are ready, but I was waiting for some comments in the issue I opened.
I saw you checked the issue yesterday, so I will rebase on top of the newest 1.2-dev and put this PR out of draft!

**Comment by 0phoff** [comment]
@Borda, I'm not sure if I should write unit tests for this change?
Honestly, this code should be covered by existing unit tests, as I just changed one line of code for another, but I couldn't really find a test that checks those code lines specifically. :shrug:

**Comment by Borda** [comment]
@0phoff it seems to.  be what I was a bit afraid of, this API did not exist back in PT 1.3, so for preserving back-compatibility we would need to use the old call, eventually make some wrapper over this function...
according to these tests against all PT versions it seems it was added in 1.4
https://github.com/PyTorchLightning/pytorch-lightning/actions/runs/468486324

File: pytorch_lightning/trainer/connectors/optimizer_connector.py [hunk 1]
```diff
@@ -65,13 +65,20 @@ def update_learning_rates(self, interval: str, monitor_metrics=None):
                             RuntimeWarning,
                         )
                         continue
+
                 # update LR
-                old_lr = lr_scheduler['scheduler'].optimizer.param_groups[0]['lr']
                 if lr_scheduler['reduce_on_plateau']:
+                    old_lr = lr_scheduler['scheduler'].optimizer.param_groups[0]['lr']
```
Comment (Borda) [inline pytorch_lightning/trainer/connectors/optimizer_connector.py:71]: well, mind remind me what is the advantage of using `get_last_lr`?
atm, I am not sure if there rea a benefit of adding this complex solution to use the last PT api
cc: @SkafteNicki @awaelchli thought?

Comment (SkafteNicki) [reply pytorch_lightning/trainer/connectors/optimizer_connector.py:71]: `old_lr` and `new_lr` are only used in the `dev_debugger` right? Are they even nessesary?

Comment (rohitgr7) [reply pytorch_lightning/trainer/connectors/optimizer_connector.py:71]: if it's doing the same thing internally, I don't see any point of updating it.
https://github.com/pytorch/pytorch/blob/c215ffb6a2de1c85a62a146a9e282908335ba3ed/torch/optim/lr_scheduler.py#L96-L99
https://github.com/pytorch/pytorch/blob/c215ffb6a2de1c85a62a146a9e282908335ba3ed/torch/optim/lr_scheduler.py#L166

**Comment by Borda** [comment]
@SkafteNicki mind decide if we proceed with this PR or drop it?

**Comment by SkafteNicki** [comment]
@Borda IMO we should drop it, atleast in its current from. The current API implicitly assumes that that all schedulers are subclass from ` torch.optim.lr_scheduler._LRScheduler` (except the special case of `ReduceLROnPlateau`), but this PR would basically open up that schedulers can be any object that implements `.step` and `get_last_lr` methods. I only think we should support official subclassed schedulers.
I am also sure this will not play nicely with something else in lightning (much likely `LearningRateMonitor`)

**Comment by Borda** [comment]
@0phoff I am closing this PR as @SkafteNicki suggested, but I am sure that your help will be very welcome...
I think that the best way would be connect on slack and talk more about what shall/can be done in this direction 🐰

**Comment by 0phoff** [comment]
@Borda I completely understand.  
This was just a problem I had with one of my own implemented LR Schedulers, but I am sure I can find a workaround.

I did send [a PR to pytorch](https://github.com/pytorch/pytorch/pull/50424), because I believe this LR Scheduler to be usefull for other people as well.
If that ever gets accepted, you might want to revisit this PR, but whether are not this will be accepted in PyTorch remains to be seen... :sweat_smile:
